src/objects/interfaces/IDiscountOffer.ts

```diff
@@ -0,0 +1,10 @@
+import { IOffer } from "./index";
+
+export interface IDiscountOffer extends IOffer {
+    // Discount percentage
+    discount: number;
+    discountID: number;
+    // Date when the discount offer expires
+    expires: string;
+    originalOffer: IOffer;
+}
```

Comment:
not sure this should extend IOffer since it also contains IOffer so data would be duplicated

Reply:
actually, the originalOffer field should be probably be removed and just leave it extending IOffer